query/src/optimizers/optimizer_expression_transform.rs

```diff
@@ -0,0 +1,199 @@
+// Copyright 2021 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::collections::HashMap;
+
+use common_datavalues::prelude::*;
+use common_exception::ErrorCode;
+use common_exception::Result;
+use common_planners::AggregatorFinalPlan;
+use common_planners::AggregatorPartialPlan;
+use common_planners::Expression;
+use common_planners::Expressions;
+use common_planners::PlanBuilder;
+use common_planners::PlanNode;
+use common_planners::PlanRewriter;
+use lazy_static::lazy_static;
+
+use crate::optimizers::Optimizer;
+use crate::sessions::DatabendQueryContextRef;
+
+pub struct ExprTransformOptimizer {}
+
+struct ExprTransformImpl {
+    before_group_by_schema: Option<DataSchemaRef>,
+}
+
+lazy_static! {
+    static ref INVERSE_OPERATOR: HashMap<&'static str, &'static str> = {
+        let mut map = HashMap::new();
+        map.insert("=", "<>");
+        map.insert("<>", "=");
+        map.insert("<", ">=");
+        map.insert("<=", ">");
+        map.insert(">", "<=");
+        map.insert(">=", "<");
+        map.insert("LIKE", "NOT LIKE");
```

Comment:
we should consider case insensitive

Reply:
https://github.com/datafuselabs/databend/pull/2045